lib/ansible/modules/net_tools/basics/uri.py

```diff
@@ -606,12 +619,21 @@ def main():
     removes = module.params['removes']
     status_code = [int(x) for x in list(module.params['status_code'])]
     socket_timeout = module.params['timeout']
+    socks5_proxy = module.params['socks5_proxy']
 
     dict_headers = module.params['headers']
 
+    if not HAVE_PYSOCKS and socks5_proxy:
+        module.fail_json(msg=missing_required_lib('PySocks'))
+
     if not re.match('^[A-Z]+$', method):
         module.fail_json(msg="Parameter 'method' needs to be a single word in uppercase, like GET or POST.")
 
+    if socks5_proxy:
+        host, port = socks5_proxy.split(':')
+        socks.set_default_proxy(socks.SOCKS5, host, port=int(port))
+        socket.socket = socks.socksocket
```

Comment:
I don't believe this should be a function of the `uri` module, but probably implemented in `ansible.module_utils.urls` instead.  However, with that being said, we wouldn't be able to use this specific line, as it could cause other issues, in that this would become a global change, and not specific to just a single request.

As such, we'd probably want to make use of `sockshandler` instead.

Reply:
That's correct. I also wanted to do that in `ansible.module_utils.urls` with use of sockshandler (as mentioned earlier) but that is not possible because `urllib` on which `ansible.module_utils.urls` uses doesn't support SOCKS handlers.

With that being said, until `urllib` supports SOCKS it is actually beneficial that this default socket redirect is scoped to a single request only.

Reply:
`urls.py` does not use `urllib`.  It uses `urllib2` on py2, and `urllib.request` on py3.

Reply:
I used `urllib` as a shortcut :) As far as I checked this doesn't change anything, or am I wrong?